### PR TITLE
Refactor FXIOS-4567 [v116] Refactor Glean in AppSettingsOptions

### PR DESCRIPTION
**Pull request by aalaaeid**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-4567)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/14567)

### Description
1. Added setUploadEnabled to the GleanWrapper class
2. Removed the import Glean from the AppSettingsOptions

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by adudenamedruby** [comment]
Build [failed on Swiftlint](https://github.com/mozilla-mobile/firefox-ios/pull/14771/checks?check_run_id=14109516789). IF you update that, it should be good. :)

File: Client/Telemetry/GleanWrapper.swift
Comment (lmarceau) [inline Client/Telemetry/GleanWrapper.swift:21]: We should remove trailing spaces as well. We use an Xcode option to help us have that automatically. Those are the options I have in my Xcode:
Screenshot 2023-06-08 at 2 34 32 PM
Screenshot 2023-06-08 at 2 34 43 PM

```suggestion

```

File: Client/Telemetry/GleanWrapper.swift
Comment (lmarceau) [inline Client/Telemetry/GleanWrapper.swift:25]:
```suggestion

```

**Review by lmarceau — COMMENTED** [review]
Little issue with Swiftlint. You can install Swiftlint locally with [homebew](https://formulae.brew.sh/formula/swiftlint) and run it in command line with `swiftlint .` at the root of the project to make sure all is good before pushing on the PR 🙌

**Comment by lmarceau** [comment]
Made the fix for swiftlint so we could unblock the PR, but seems there's some changes needed in the unit tests target. Could you look into this 🙏 ? `MockGleanWrapper` should conform to the protocol changes.

Screenshot 2023-06-08 at 2 51 24 PM

**Review by adudenamedruby — APPROVED** [review]
@lmarceau I'll let you approve as well, just to keep track of things.

**Review by lmarceau — APPROVED** [review]

**Pull request merged**
